### PR TITLE
Register account

### LINKED ISSUE #5: Register user account screen
As a user I would like to be able to register a user account.
When registering a user account e-mail address (= user name), a last name and a first name, an address, a mobile phone number and a password.

ESTIMATE: 5

### PR DESCRIPTION
**Pull request by LucasStrand**
Fixes #5 
# Description

Added a new view for register user account with validations for each field.

## Lists of changes

- [x] Added a new view and a viewcontroller
- [x] Added validations 

image

## Testing

Short summary of what should be tested
- [ ] Enter invalid inputs and see if an error message prompts

Known issues: 
- I'm well aware the mockup shows two inputs for the password but currently SwiftUI has a bug which messes the page up if you have two SecureFields.
- We need to connect all the views together
- There are currently no backend connected, we probably need a card for it(if we do not already have)

### CODE REVIEW

**Comment by LucasStrand** [comment]
I'm also aware of that i'm breaking clean code standard by commenting code instead of removing it, but as soon as SwiftUI isn't bugged, we'll need the code

File: FlexiCharge/FlexiCharge/Views/RegisterAccount/RegisterAccountView.swift [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                    ZStack(alignment: .leading){
+                        TextField("Email", text: $email, onEditingChanged: { changed in
+                            if changed{
+                                /*
+                                if !validateEmail(email: email){
+                                    validEmail = false
+                                }else{
+                                    validEmail = true
+                                }*/
+                            }
```
Comment (filipfloden) [inline FlexiCharge/FlexiCharge/Views/RegisterAccount/RegisterAccountView.swift:48]: Does the `onEditingChanged` still serve a purpose here or is it all in the `validateInputs` further down?

File: FlexiCharge/FlexiCharge/Views/RegisterAccount/RegisterAccountView.swift [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                    .padding(.top)
+                    Spacer()
+                    /*----------Register button and the following text----------*/
+                    VStack{
+                        Text(validationText)
+                            .foregroundColor(.red)
+                            .padding(.bottom)
+                        Button(action: {
+                                validationText = validateInputs(email: email, mobileNumber: mobileNumber, password: password, TOSCheckBox: tosCheckBox)
+
```
Comment (filipfloden) [inline FlexiCharge/FlexiCharge/Views/RegisterAccount/RegisterAccountView.swift:135]: I'm assuming that this should then actually 'register' the user if all validations are fine?

**Review by filipfloden — CHANGES_REQUESTED** [review]

File: FlexiCharge/FlexiCharge/Views/RegisterAccount/RegisterAccountView.swift (hunk 2 above)
Comment (LucasStrand) [reply FlexiCharge/FlexiCharge/Views/RegisterAccount/RegisterAccountView.swift:135]: Yes that is indeed correct 👍🏻 As soon as we have access to the database, the actual registration be called here.

**Review by filipfloden — APPROVED** [review]

**Pull request merged**